### DISCUSSION TITLE
Split into services (instead of / additional to pods)

### DISCUSSION THREAD

**Discussion by jochen-schuettler** [sustainable-computing-io/kepler#1241] [Ideas]
Given the standard kepler exporter dashboard we can visualize energy consumption by component (PKG, DRAM, OTHER) or by pod and component. The pods have names like 'service-replicaset-pod / namespace'. I'd like another view that adds the consumption of all pods of a each service, i.e. with names like 'service1*', 'service2*', component-wise. Such service-based consumption curves would improve visibility e.g. of scaling effects.
I can add such a frame myself in the GUI, but that is time-consuming and error-prone, and it's valid only as long as the replicaset id doesn't change.
So I'd like a programmatic version, which could be created in kepler, get exported, and visualized in the standard Kepler exporter dashboard.

**Comment by marceloamaral** [comment]
Due to scalability concerns, we avoid adding extra labels to the Kepler metrics to mitigate Prometheus overhead. Introducing new labels has the potential to significantly increase the number of exported metrics (the metric cardinality), which could impact performance.

However, metrics can be aggregated using [Prometheus rules](https://prometheus.io/docs/prometheus/latest/querying/operators/#many-to-one-and-one-to-many-vector-matches) to combine different metrics. In cases where a label is absent, you can leverage the label-exporter tool available at https://github.com/sustainable-computing-io/label-exporter or https://github.com/kubernetes/kube-state-metrics/blob/main/docs/pod-metrics.md to assist with this process.

Than, you can merge the metric labels as follows:

```
kepler_container_platform_joules_total  * on (container_name, pod_name) group_left(service,foo,bar) kube_pod_owner

or

kepler_container_platform_joules_total  * on (container_name, pod_name) group_left(foo,bar,baz) kube_pod_labels
```

**Comment by jochen-schuettler** [comment]
I'll try that, please leave open for further questions.

**Comment by jochen-schuettler** [comment]
It turned out, the solution was even easier: use
`sum by(container_name) (irate(...))`
in Grafana and change Options->Legend to
`{{container_name}} / {{container_namespace}} `.


So, thx for your help.
